### PR TITLE
refactor: Select and Include type generation

### PR DESCRIPTION
**Pull request by SevInf**
Split type generation from `Model.ts` and use ts-builders for
generation. Adds several helpers, necessary for this refactoring:
`typeDeclaration`, `export` and `keyType`.

### CODE REVIEW

File: packages/client/src/__tests__/integration/happy/exhaustive-schema-mongo/__snapshots__/test.ts.snap [hunk 1]
```diff
@@ -1736,9 +1736,7 @@ export namespace Prisma {
   }
 
 
-  export type EmbedInclude = {
-
-  } 
+  export type EmbedInclude = {}
```
Comment (SevInf) [inline packages/client/src/__tests__/integration/happy/exhaustive-schema-mongo/__snapshots__/test.ts.snap:1739]: Whitespace changes only

File: packages/client/src/generation/ts-builders/KeyType.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { BasicBuilder } from './BasicBuilder'
+import { FunctionType } from './FunctionType'
+import { ObjectType } from './ObjectType'
+import { UnionType } from './UnionType'
+import { Writer } from './Writer'
+
+export class KeyType implements BasicBuilder {
+  constructor(public baseType: AnyTypeBuilder, public key: string) {}
+  write(writer: Writer): void {
+    const needsBraces =
```
Comment (aqrln) [inline packages/client/src/generation/ts-builders/KeyType.ts:11]: Should this be something like `hasBraces` or `needsParens` instead?

Also, would it make sense to define this as a method in `BasicBuilder` interface and let each type tell if it needs to be wrapped when indexing, or is not worth it and not going to be useful elsewhere?

File: packages/client/src/generation/ts-builders/TypeDeclaration.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  expect(stringify(decl)).toMatchInlineSnapshot(`
+    /**
+     * Type for stuff
+     */
+    type B = A
+  `)
+})
+
+test('with generic parameters', () => {
+  const decl = typeDeclaration('B', A).addGenericParameter(genericParameter('T'))
```
[the remark this answers is not in the document]
Comment (SevInf) [reply packages/client/src/generation/ts-builders/TypeDeclaration.test.ts:24]: Added.

File: packages/client/src/generation/ts-builders/KeyType.ts (hunk 2 above)
Comment (SevInf) [reply packages/client/src/generation/ts-builders/KeyType.ts]: Thanks, did that in the next commit.

**Review by aqrln — APPROVED** [review]
💯

**Pull request merged**